packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -225,85 +227,35 @@ void updateLocalProperties({String projectPath, BuildInfo buildInfo}) {
-    final IOSink sink = appGradleTmp.openWrite();
-
-    inputStream.transform(utf8.decoder)
-        .transform(const LineSplitter())
-        .map((String line) {
-
-          // find and replace build number
-          if (buildInfo.buildNumber != null) {
-            if (line.contains(new RegExp(r'^[ |\t]*(versionCode)[ =\t]*\d*'))) {
-              return line.splitMapJoin(new RegExp(r'(versionCode)[ =\t]*\d*'), onMatch: (Match m) {
-                return 'versionCode ${buildInfo.buildNumber}';
-              });
-            }
-          }
+  final String buildName = buildInfo?.buildName ?? manifest?.buildName;
+  if (buildName != null) {
+    settings.values['flutter.versionName'] = buildName;
+    changed = true;
+  }
 
-          // find and replace build name
-          if (buildInfo.buildName != null) {
-            if (line.contains(new RegExp(r'^[ |\t]*(versionName)[ =\t]*\"[0-9.]*"'))) {
-              return line.splitMapJoin(new RegExp(r'(versionName)[ =\t]*\"[0-9.]*"'), onMatch: (Match m) {
-                return 'versionName "${buildInfo.buildName}"';
-              });
-            }
-          }
-          return line;
-        })
-        .listen((String line) {
-            sink.writeln(line);
-          },
-          onDone: () {
-            sink.close();
-            try {
-              final File gradleOld = appGradle.renameSync(fs.path.join('android', 'app', 'build.gradle.old'));
-              appGradleTmp.renameSync(fs.path.join('android', 'app', 'build.gradle'));
-              gradleOld.deleteSync();
-              completer.complete();
-            } catch (error) {
-              printError('Failed to change version properties. $error');
-              completer.completeError('Failed to change version properties. $error');
-            }
-          },
-          onError: (dynamic error, StackTrace stack) {
-            printError('Failed to change version properties. ${error.toString()}');
-            sink.close();
-            appGradleTmp.deleteSync();
-            completer.completeError('Failed to change version properties. ${error.toString()}', stack);
-          },
-    );
+  final int buildNumber = buildInfo?.buildNumber ?? manifest?.buildNumber;
+  if (buildNumber != null) {
+    settings.values['flutter.versionCode'] = '$buildNumber';
+    changed = true;
   }
 
-  return completer.future;
+  if (changed)
+    settings.writeContents(localProperties);
 }
 
 Future<Null> buildGradleProject(BuildInfo buildInfo, String target) async {
   // Update the local.properties file with the build mode.
   // FlutterPlugin v1 reads local.properties to determine build mode. Plugin v2
   // uses the standard Android way to determine what to build, but we still
   // update local.properties, in case we want to use it in the future.
-  updateLocalProperties(buildInfo: buildInfo);
-  await findAndReplaceVersionProperties(buildInfo: buildInfo);
+  await updateLocalProperties(buildInfo: buildInfo);
```

Comment:
The comment above is not accurate anymore. We should state that the Gradle script reads `local.properties` to learn build name and number, in case they are being overwritten by the CLI invocation.